zilliqa/src/db.rs

```diff
@@ -221,6 +235,111 @@ impl Db {
         Ok(())
     }
 
+    pub fn create_checkpoint_path(&self, height: u64) -> Result<Option<Box<Path>>> {
+        let Some(base_path) = &self.path else {
+            // If we don't have on-disk persistency, disable checkpoints too
+            return Ok(None);
+        };
+        let snapshot_path = base_path.join("snapshots").join(height.to_string());
+        fs::create_dir_all(&snapshot_path)?;
+        Ok(Some(snapshot_path.into_boxed_path()))
+    }
+
+    pub fn load_trusted_checkpoint<P: AsRef<Path>>(
+        &self,
+        path: P,
+        hash: &Hash,
+        our_shard_id: u64,
+    ) -> Result<Block> {
+        let input = File::open(path)?;
+        let reader = BufReader::with_capacity(8192 * 1024, input); // 8 MiB read chunks
```

Comment:
Is there a reason we picked 8 MiB? The default is 8 KiB - Was that bad for some reason?

Reply:
If we're potentially going to be reading/writing multi-GB files, I think a slightly larger batch size is warranted. The only tradeoff is memory (to my understanding at least...) and 8MiB seems like perfectly reasonable usage, and in return we will have 1000x fewer I/O syscalls when exporting and importing huge files.